src/Engine/MapController.java

```diff
@@ -103,7 +107,8 @@ else if(Environment.Terrains.FACTORY == loc.getEnvironment().terrainType
 				System.out.println("found a factory");
 				// TODO: Don't hard-code this. Also, is DamageChart the best place for UnitEnum?
 				DamageChart.UnitEnum[] units = {DamageChart.UnitEnum.INFANTRY};
-				actionMenu = new GameMenu(units);
+				String[] labels = {"Infantry: " + myGame.activeCO.getUnitModel(DamageChart.UnitEnum.INFANTRY).moneyCost};
```

Comment:
I don't like hard-coding this here. MapView can get the string from the Enum value itself, and the cost by querying the activeCO in the GameInstance.